src/coreclr/vm/invokeutil.cpp

```diff
@@ -124,191 +124,115 @@ void *InvokeUtil::GetIntPtrValue(OBJECTREF pObj) {
     CONTRACTL {
         THROWS;
         GC_NOTRIGGER; // Caller does not protect object references
         MODE_COOPERATIVE;
         PRECONDITION(!th.IsNull());
-        PRECONDITION(CheckPointer(pObjUNSAFE));
         INJECT_FAULT(COMPlusThrowOM());
     }
     CONTRACTL_END;
 
     void *pArgDst = argDest->GetDestinationAddress();
+    PVOID *rArg = *pArgRef;
+    CorElementType type = th.GetVerifierCorElementType();
 
-    OBJECTREF rObj = *pObjUNSAFE;
-    MethodTable* pMT;
-    CorElementType oType;
-    CorElementType type;
-
-    if (rObj != 0) {
-        pMT = rObj->GetMethodTable();
-        oType = pMT->GetInternalCorElementType();
-    }
-    else {
-        pMT = 0;
-        oType = ELEMENT_TYPE_OBJECT;
-    }
-    type = th.GetVerifierCorElementType();
-
-    // This basically maps the Signature type our type and calls the CreatePrimitiveValue
-    //  method.  We can omit this if we get alignment on these types.
     switch (type) {
     case ELEMENT_TYPE_BOOLEAN:
-    case ELEMENT_TYPE_I1:
+    case ELEMENT_TYPE_CHAR:
     case ELEMENT_TYPE_U1:
+    case ELEMENT_TYPE_I1:
+        _ASSERTE(rArg != NULL);
+        *(PVOID *)pArgDst = (INT8*)*rArg;
+        break;
+
     case ELEMENT_TYPE_I2:
     case ELEMENT_TYPE_U2:
-    case ELEMENT_TYPE_CHAR:
+        _ASSERTE(rArg != NULL);
+        *(PVOID *)pArgDst = (INT16*)*rArg;
+        break;
+
     case ELEMENT_TYPE_I4:
     case ELEMENT_TYPE_U4:
     case ELEMENT_TYPE_R4:
-    IN_TARGET_32BIT(case ELEMENT_TYPE_I:)
     IN_TARGET_32BIT(case ELEMENT_TYPE_U:)
-    {
-        // If we got the univeral zero...Then assign it and exit.
-        if (rObj == 0)
-            *(PVOID *)pArgDst = 0;
-        else
-        {
-            ARG_SLOT slot;
-            CreatePrimitiveValue(type, oType, rObj, &slot);
-            *(PVOID *)pArgDst = (PVOID)slot;
-        }
+    IN_TARGET_32BIT(case ELEMENT_TYPE_I:)
+        _ASSERTE(rArg != NULL);
+        *(PVOID *)pArgDst = (INT32*)*rArg;
         break;
-    }
-
 
     case ELEMENT_TYPE_I8:
     case ELEMENT_TYPE_U8:
     case ELEMENT_TYPE_R8:
     IN_TARGET_64BIT(case ELEMENT_TYPE_I:)
     IN_TARGET_64BIT(case ELEMENT_TYPE_U:)
     {
-        // If we got the univeral zero...Then assign it and exit.
-        if (rObj == 0)
-            *(INT64 *)pArgDst = 0;
-        else
-        {
-            ARG_SLOT slot;
-            CreatePrimitiveValue(type, oType, rObj, &slot);
-            *(INT64 *)pArgDst = (INT64)slot;
-        }
+        _ASSERTE(rArg != NULL);
+        *(INT64 *)pArgDst = (INT64)*rArg;
         break;
     }
 
     case ELEMENT_TYPE_VALUETYPE:
     {
-        // If we got the universal zero...Then assign it and exit.
-        if (rObj == 0) {
-            InitValueClassArg(argDest, th.AsMethodTable());
-         }
-        else {
-            if (!th.AsMethodTable()->UnBoxIntoArg(argDest, rObj))
-                COMPlusThrow(kArgumentException, W("Arg_ObjObj"));
+        {
```

Comment:
One level of indentation can be removed here.

Reply:
Yep. Thanks